File: CS50x/Problem Set 2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

float calc(string text);

int main(void){
    string text = get_string("Text: ");
    int cl_index = round(calc(text));
    if (cl_index < 1){
        printf("Before Grade 1\n");
    }
    else if (cl_index > 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", cl_index);
    }
}

float calc(string text){
    int len = strlen(text);
    int letters = 0;
    int sentences = 0;
    int words = 1;
    for(int i = 0; i < len; i++){
        if(isalpha(text[i])){
            letters += 1;
        }
        else if(isspace(text[i])){
            words += 1;
        }
        else if((int)text[i] == 33 || (int)text[i] == 46 || (int)text[i] == 63){
            sentences += 1;
        }
    }
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;
    return 0.0588 * l - 0.296 * s - 15.8;
}

File: CS50x/Problem Set 6/sentimental-readability/readability.py
import sys

def main():
    text = input("Text: ").strip()
    letters = 0.00
    words = 0.00
    sentences = 0.00
    pause = False
    for char in text:
        if char.isalnum():
            letters += 1
            pause = False
        elif char == " " and pause == False:
            words += 1
        elif char == ",":
            words += 1
            pause = True
        elif char in [".", "!", "?"]:
            words += 1
            sentences += 1
            pause = True
    coleman = round(0.0588 * (100 * (letters/words)) - 0.296 * (100 * (sentences/words)) - 15.8)
    if coleman < 1:
        print("Before Grade 1")
        sys.exit(0)
    elif coleman >= 16:
        print("Grade 16+")
        sys.exit(0)
    print(f"Grade {coleman}")


if __name__ == "__main__":
    main()
